packages/block-library/src/group/variations.js

```diff
@@ -9,9 +9,9 @@ const variations = [
 		name: 'group',
 		title: __( 'Group' ),
 		description: __( 'Gather blocks in a container.' ),
-		attributes: { layout: { type: 'constrained' } },
+		attributes: { layout: { type: 'constrained', isDefault: true } },
```

Comment:
`isDefault: true` is the magic flag that allows us to know whether a block is in its "default" state, that is, recently inserted with no layout selected. 

I'm using it in combination with `hasInnerBlocks` to know when to show the placeholder.

E.g., `const showPlaceholder = isDefault && ! hasInnerBlocks;`.

When a layout is selected in the editor, the `isDefault` attribute is stripped so it doesn't appear in the markup, at least if added in the editor.

I'm not yet confident this is the right way to go about things as it adds an attribute by default to group block that meaningless outside the context of the editor. 

Maybe there's another way, or maybe what I'm doing is fine. 

I'll keep testing and try to come up with something.

I'll try to think of an alternative.

Reply:
That seems a bit fragile to me because if unchanged, that value is going to persist in the post. Why are we not showing the placeholder state for all group blocks (not row or stack) without any inner blocks?

Reply:
I agree with Riad. I thought an alternative would be to just check if it has innerBlocks and apply some classes, instead of [only-child styles](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/group/editor.scss#L27), when we have selected something in the placeholder and still don't have any innerBlocks. This way we would have the `revert to placeholder` behavior discussed [here](https://github.com/WordPress/gutenberg/issues/43433#issuecomment-1257872806):

>If you just click "Group" in the flow illustrated, you should get the group as it exists today, i.e. with the inserter. If you then don't add any blocks inside and deselect, we can revert back to the setup state as shown. Does that make sense?

I haven't tried it out though..

Reply:
Thanks for the feedback. My instinct was warning me something wasn't right, so I appreciate the confirmation, folks.

> Why are we not showing the placeholder state for all group blocks (not row or stack) without any inner blocks?

What I was attempting to avoid, perhaps unnecessarily and despite @jasmussen's [comments](https://github.com/WordPress/gutenberg/issues/43433#issuecomment-1257872806), was the situation in which a user:

1. Inserts a group block
2. Selects a variation, but does not add any child blocks
3. Works a bit, saves.
4. Returns to the page to find a group variation placeholder
5. Wonders why they have to select a variation again

I'll remove the `isDefault` attribute so folks can test. Perhaps I was, as usual, overthinking it.

Reply:
> check if it has innerBlocks and apply some classes, instead of only-child styles, when we have selected something in the placeholder and still don't have any innerBlocks

Do you mean create a new classname to indicate that a variation has been selected, regardless of whether there are innerBlocks or not?

That would be an alternative to the `isDefault` flag. It would also create a persistent value. 

It's worth pursuing. I'll try it out. Thanks!

🍺

Reply:
> What I was attempting to avoid, perhaps unnecessarily and despite @jasmussen's https://github.com/WordPress/gutenberg/issues/43433#issuecomment-1257872806, was the situation in which a user:

I withdraw that. This is the assumption I was going on, from the [comments](https://github.com/WordPress/gutenberg/pull/43496#issuecomment-1298249501) below:

> Since this new state is mainly intended to be shown for an empty group inserted fresh,

Reply:
> Why are we not showing the placeholder state for all group blocks (not row or stack) without any inner blocks?

If we show the placeholder for Groups without inner blocks, or even without a variation selected, we'll lose the possibility of using Groups for visual/aesthetic purposes, such as [this](https://block-museum.com/2022/01/26/026/). We need to check for the existence of any custom styling before showing the placeholder.

Reply:
> If we show the placeholder for Groups without inner blocks, or even without a variation selected, we'll lose the possibility of using Groups for visual/aesthetic purposes, such as [this](https://block-museum.com/2022/01/26/026/). We need to check for the existence of any custom styling before showing the placeholder.

Also a great line of reasoning, thanks @tellthemachines 

Maybe a utility class or other HTML attribute is called for – `is-empty` or `aria-placeholder,="Empty Group block"` – to replace the `isDefault` block attribute and work in conjunction with the placeholder as a flag to show that a variation hasn't been selected.

I'll experiment and seek feedback.

Reply:
> Why are we not showing the placeholder state for all group blocks (not row or stack) without any inner blocks?

Also, I forgot say, the default state of a Group block has no inner blocks. So even if a user selects the default variation, we need something else to tell the editor to remove the placeholder since it'll still have no inner blocks.